pkg/console/operator/sync_v400.go

```diff
@@ -556,13 +556,75 @@ func (co *consoleOperator) SyncTrustedCAConfigMap(ctx context.Context, operatorC
-	// validate first, to avoid a broken volume mount & a crashlooping console
-	okToMount, reason, err = co.ValidateCustomLogo(ctx, operatorConfig)
+func (co *consoleOperator) SyncCustomLogoConfigMaps(ctx context.Context, operatorConfig *operatorv1.Console) ([]configmapsub.CustomLogoRef, string, error) {
+	configMapsToSync := getConfigMapsToSync(operatorConfig)
+	var (
+		aggregatedError error
+		err             error
+		reason          string
+	)
+	for _, logoToSync := range configMapsToSync {
+		logoToSync.OkToMount, reason, err = co.SyncCustomLogoConfigMap(ctx, logoToSync)
+		if err != nil {
+			if aggregatedError == nil {
+				aggregatedError = fmt.Errorf("One or more errors were encountered while configuring custom logos:\n\t- %v, %s", logoToSync, err.Error())
+			} else {
+				aggregatedError = fmt.Errorf("%s\n\t- %v, %s", aggregatedError.Error(), logoToSync, err.Error())
+			}
+		}
+	}
+	if aggregatedError != nil {
+		return nil, reason, aggregatedError
+	}
+	return configMapsToSync, "", nil
+}
 
-	if okToMount || configmapsub.IsRemoved(operatorConfig) {
-		if err := co.UpdateCustomLogoSyncSource(operatorConfig); err != nil {
-			return false, "FailedSyncSource", customerrors.NewCustomLogoError("custom logo sync source update error")
+func getConfigMapsToSync(operatorConfig *operatorv1.Console) []configmapsub.CustomLogoRef {
+	var normalizedLogos = []configmapsub.CustomLogoRef{}
+	var skipDeprecatedLogoSync = false
+	if operatorConfig.Spec.Customization.CustomLogos != nil {
+		for _, logo := range operatorConfig.Spec.Customization.CustomLogos {
+			if logo.Type == operatorv1.LogoTypeMasthead {
+				skipDeprecatedLogoSync = true
+			}
+			for _, theme := range logo.Themes {
+				normalizedLogos = append(
+					normalizedLogos,
+					configmapsub.CustomLogoRef{
+						Location: logo.Type,
+						Theme:    theme.Type,
+						File:     theme.File,
+					},
+				)
+			}
+		}
+	}
+
+	if !skipDeprecatedLogoSync && &operatorConfig.Spec.Customization.CustomLogoFile != nil {
+		normalizedLogos = append(
+			normalizedLogos,
+			configmapsub.CustomLogoRef{
+				Location: operatorv1.LogoTypeMasthead,
+				Theme:    operatorv1.ThemeTypeDefault,
+				File:     operatorConfig.Spec.Customization.CustomLogoFile,
+			},
+		)
+	}
+
+	return normalizedLogos
+}
+
+// TODO CONSOLE-4061 - update this function to handle new custom favicon API. Will need to handle
+// multiple configmaps.
+// - Validate existence of custom icon configmaps defined in the console operator config,
+// - populate to the console config
+// - create volumes/volumemounts in console deployment
+func (co *consoleOperator) SyncCustomLogoConfigMap(ctx context.Context, customLogo configmapsub.CustomLogoRef) (bool, string, error) {
+	// validate first, to avoid a broken volume mount & a crashlooping console
+	okToMount, reason, err := co.ValidateCustomLogo(ctx, customLogo.File)
+	if okToMount {
```

Comment:
I think we can remove the `okToMount` field from the `configmapsub.CustomLogoRef` since we either mount all the CMs or none...

Reply:
in which case here we will check just for `err`